chart/templates/_helpers.tpl

```diff
@@ -205,3 +205,28 @@ Usage:
         {{- end -}}
     {{- end -}}
 {{- end -}}
+
+{{/*
+    Generate the default StorageClass parameters.
+    This is required because StorageClass parameters cannot be patched after creation.
+    If the StorageClass already exists, the default StorageClass carries the parameters and values
+    of that StorageClass. Else, it carries the default parameters and values.
+*/}}
+{{- define "storageClass.parameters" -}}
+    {{- $scName := index . 0 -}}
+    {{- $valuesRepl := index . 1 -}}
+    {{- $valuesProtocol := index . 2 -}}
+
+    {{- $sc := lookup "storage.k8s.io/v1" "StorageClass" "" $scName -}}
+    {{- if $sc -}}
+        {{ range $param, $val := $sc.parameters }}
+{{ $param | quote }}: {{ $val | quote }}
+        {{- end -}}
+    {{- else -}}
+{{/*
+    Latest defaults
+*/}}
+protocol: {{ $valuesProtocol | quote }}
```

Comment:
this we could take from values.yaml and avoid modifying the template again in the future

Reply:
Good one, made this change.